src/Magento/FunctionalTestingFramework/DataGenerator/Objects/EntityDataObject.php

```diff
@@ -177,15 +177,22 @@ public function getDataByName($name, $uniquenessFormat)
 
         $name_lower = strtolower($name);
 
-        if ($this->data !== null && array_key_exists($name_lower, $this->data)) {
+        if ($this->data === null) {
+            return null;
+        }
+
+        if (array_key_exists($name_lower, $this->data)) {
```

Comment:
`$this->data` can still be camelCased, so you need to compare `lower_name` === `lower_index`.
Could leverage `array_change_key_case()` for a temp array if needed

Reply:
I don't quite understand.

The second branch of this conditional `elseif (array_key_exists($name, $this->data))` catches the cases where the key is camelCased.

Reply:
Honestly I'm not sure anymore either. I was running the attached usecase to the ticket and it was causing mis-matching issues.
Reran to get some proof and it's no longer failing. So disregard this comment.

Reply:
Haha cool. I was able to run the usecase attached to the ticket successfully and with my included unit test I think we're safe.